contracts/src/protocol/OmniAVS.sol

```diff
@@ -0,0 +1,227 @@
+    }
+
+    /// @dev Override ServiceManagerBase.deregisterOperatorFromAVS, to track list of operators
+    ///      We need to track list of operators so we can call registryCoordinator.updateOperators(operators);
+    ///      before syncing with omni.
+    function deregisterOperatorFromAVS(address operator) public override {
+        ServiceManagerBase.deregisterOperatorFromAVS(operator);
+        _removeOperator(operator);
+    }
+
+    /**
+     * Admin controls
+     */
+
+    /// @inheritdoc IOmniAVSAdmin
+    function setOmniPortal(IOmniPortal omni_) external onlyOwner {
+        omni = omni_;
+    }
+
+    /// @inheritdoc IOmniAVSAdmin
+    function setOmniChainId(uint64 omniChainId_) external onlyOwner {
+        omniChainId = omniChainId_;
+    }
+
+    /**
+     * View functions
+     */
+
+    /// @inheritdoc IOmniAVS
+    function getValidators() public view returns (Validator[] memory) {
+        return getValidators(block.number);
+    }
+
+    /// @inheritdoc IOmniAVS
+    function getValidators(uint256 blockNumber) public view returns (Validator[] memory validators) {
+        /// we only provide on quorum, so we only need the first element
+        Operator[] memory ops = _getOperatorState(blockNumber)[0];
+
+        // We translate OperatorStateRetriever.Operator[] to Validator[], splitting Operator.stake into:
+        //  - Validator.delegated (total amount delegated to the operator) and
+        //  - Validator.staked (total amount staked by the operator)
+
+        validators = new Validator[](ops.length);
+
+        for (uint256 i = 0; i < ops.length; i++) {
+            address addr = ops[i].operator;
+            uint96 staked = _getStaked(addr);
+            uint96 delegations = ops[i].stake - staked;
+            validators[i] = Validator(addr, delegations, staked);
+        }
+
+        return validators;
+    }
+
+    /**
+     * Internal view functions
+     */
+
+    /// @dev Read operator state from avs registries, at a specific block number
+    function _getOperatorState(uint256 blockNumber) public view returns (Operator[][] memory) {
+        return OperatorStateRetriever.getOperatorState(_registryCoordinator, QUORUM_NUMBERS, uint32(blockNumber));
+    }
+
+    /// @dev Returns the total amount staked by the operator, not including deletations
+    ///      This requires us translate the operators active delegatable shares into strategyParam
+    ///      weigted steake amount, for QUORUM_NUMBER.
+    function _getStaked(address operator) internal view returns (uint96) {
+        IStakeRegistry.StrategyParams[] memory strategyParams = _strategyParams();
+        (IStrategy[] memory strategies, uint256[] memory shares) =
+            DelegationManager(address(_delegationManager)).getDelegatableShares(operator);
+
+        uint96 staked;
+
+        for (uint256 i = 0; i < strategies.length; i++) {
+            IStrategy strat = strategies[i];
+            uint256 sharesAmt = shares[i];
```

Comment:
this is only the restaked amount for the operator themselves?

Reply:
for that strategy